src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -1341,31 +1340,31 @@ public enum Dataset
         /// <summary>
         /// The options for the <see cref="ImageClassificationTransformer"/>.
         /// </summary>
-        internal sealed class Options : TransformInputBase
+        public sealed class Options : TransformInputBase
         {
-            /// <summary>
-            /// Location of the TensorFlow model.
-            /// </summary>
-            [Argument(ArgumentType.Required, HelpText = "TensorFlow model used by the transform. Please see https://www.tensorflow.org/mobile/prepare_models for more details.", SortOrder = 0)]
-            public string ModelLocation;
-
             /// <summary>
             /// The names of the model inputs.
             /// </summary>
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 1)]
+            [Argument(ArgumentType.Multiple , HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 1)]
             public string[] InputColumns;
 
             /// <summary>
             /// The names of the requested model outputs.
             /// </summary>
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 2)]
+            [Argument(ArgumentType.Multiple , HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 2)]
             public string[] OutputColumns;
 
+            /// <summary>
+            /// The names of the model input features.
+            /// </summary>
+            [Argument(ArgumentType.AtMostOnce | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "features", SortOrder = 1)]
+            public string FeaturesColumnName;
+
             /// <summary>
             /// The name of the label column in <see cref="IDataView"/> that will be mapped to label node in TensorFlow model.
             /// </summary>
-            [Argument(ArgumentType.AtMostOnce, HelpText = "Training labels.", ShortName = "label", SortOrder = 4)]
-            public string LabelColumn;
+            [Argument(ArgumentType.AtMostOnce | ArgumentType.Required, HelpText = "Training labels.", ShortName = "label", SortOrder = 4)]
+            public string LabelColumnName;
```

Comment:
What's the difference between this and `TensorFlowLabel`?

Reply:
As far as I can understand, TensorFlowLabel is the name of the TF node in the model that contains the label, while the LabelColumn is the name of the column in the IDataView. They are assigned the same value, so working on removing one of them.

---
In reply to: [337697817](https://github.com/dotnet/machinelearning/pull/4362#discussion_r337697817) [](ancestors = 337697817)